### ISSUE TITLE
Bar showing up on main page

### ISSUE THREAD

**Issue by cbattlegear** [reddit-archive/reddit-companion#5]
The bar is showing up on my reddit main page. For some reason it keeps bringing up this article: http://www.reddit.com/r/lounge/comments/cp68g/what_the_plebes_are_reading_these_days/

It looks like so: 

http://i.imgur.com/DcQdd.png

I am running OS X with RES and Chrome 14.0.803.0 dev

**Comment by sleelin** [comment]
I also get this problem, tried in both Ubuntu and Windows 7 (Chrome 12), however it is directing me at an r/firstworldproblems submission.

Any chance you could just block all of reddit.com? (not really needed on self posts)

**Comment by chromakode** [comment]
Another solution would be to remember when you've closed the bar using "X" and not auto-show it again for the page. Would this work for you?

**Comment by sleelin** [comment]
Yes it would, so long as there was some way to undo that removal, maybe in the options.html page when you implement that (but I can hold out with not being able to undo the removal until you implement that).

Just out of curiosity, whereabouts/how do you check whether the link has been submitted, and then show the bar if it has been? I was snooping around the code thinking I might have been able to fork you and solve this myself, but just had no idea where it was O.o

**Comment by cbattlegear** [comment]
Yeah, being able to block out certain domains would be very beneficial. Still curious why it is grabbing a seemingly random article when you land on www.reddit.com.

**Comment by chromakode** [comment]
This should only happen if you clicked the alien icon when viewing the front page. The extension queries (searches) reddit for the most relevant post with the URL you're currently on. The post you see in the bar is whatever the reddit search returned. Unfortunately, this is currently sticky and will keep popping up when you revisit the page.

@sleelin, each tab connects a Port to the background page when it loads. When the Port connects, the background page checks for reddit information. You can find this on line 401 of background.js:

``` javascript
  switch (name) {
    case 'overlay':
      tabStatus.add(port)
      var tab = port.sender.tab,
          info = setPageActionIcon(tab)
      if (info) {
        console.log('Recognized page '+tab.url, info)
        tabStatus.showInfo(tab.id, info.name)
      }
      break
```

**Comment by sleelin** [comment]
Oh yeah, I see now. I uninstalled the plugin and installed from source just then and the tile bar above reddit.com went away until i clicked on the reddit alien in the omnibar again (after which it continued to show up until I removed and reinstalled the plugin again). Since you can reload the bar by clicking on the reddit alien, something in options.html would be redundant. Where does it store information about already viewed pages?

Thanks for pointing that out, in my search I traced to that point already, but couldn't work out where to from there :(

In the mean time, must... resist... urge to push alien icon...

**Comment by mhweaver** [comment]
> Where does it store information about already viewed pages?

It looks like previously viewed pages is stored in redditInfo.url (check out background.js) by redditInfo.setURL()

**Comment by mhweaver** [comment]
[Pull request](https://github.com/reddit/reddit-companion/pull/11) submitted as a possible approach to fix this issue.

**Comment by chromakode** [comment]
Fixed in 400528d99411ba6566beedb4296609ba7324261a.
